### DISCUSSION TITLE
Add winding order option to PxTriangleMeshDesc

### DISCUSSION THREAD

**Discussion by kareemkermad** [NVIDIA-Omniverse/PhysX#203] [Q&A]
My meshes are stored with indices in the opposite winding order that PhysX expects. Would it be possible to change the triangle `stride` type to a signed integer, so that I can specify a negative value for the stride? Or, alternatively, add a flag to specify the winding order. This will allow me to avoid allocating and reversing the array of indices, which is quite costly.

**Comment by nvtw** [comment]
Hi, I don't think that allowing a negative integer for the stride will solve the problem. The stride defines how much the reading pointer gets advanced when moving from one triangle to the next. A negative stride would mean the the pointer would read the triangles from right to left. This could in theory reverse the order of the triangles but not the winding because a single triangle is still read as before. And reading usually starts at 0, so a negative stride would need deeper changes to work.
I would recommend to use a temporary array to switch the winding, even if that's not for free. Keep in mind that you can store the cooked triangle meshes in files such that loading becomes very fast and the cooking overhead can be completely avoided (and therefore also the need for reversing the winding) once the cooked files are available.
